### ISSUE TITLE
Update the way we are indexing ContentBase models

### ISSUE THREAD

**Issue by Ravenstine** [SCPR/SCPRv4#460]
Even if Sean isn't here until Monday, I think this is worth doing just to further validate the ideas on using ContentBase more efficiently.  I'd probably think of this more as a PoC, for now.

AC: 

- [x] Index related content with Segments.
- [x] Index "published_content" (aka rundown content) along with episodes.
- [x] Check for any performance improvement? ¯\_(ツ)_/¯  
- [x] Figure out method to re-index all articles.

**Comment by Ravenstine** [comment]
Some other things that I found during last sprint that might be incorporated into this ticket:

- in `views/programs/kpcc/_featured_segment`, it looks like we are using the original object to render the bylines, and similarly in `single`.  I think there's an instance where we are using joined_bylines from the original_object, but I misplaced where that was at the moment.  This might be due to a shortcoming in the `attributions` on an Article but, if that's the case, we need to find a way around this.
- content_carousel needs to know the class of its content, for some reason.  But we're going to get rid of this anyway, so...?
- `verticals/shared/_lead` is another instance where we should switch to getting related content from ES.
- `app/views/pmp/story.html.erb` has an example of why we might want to consider indexing the "is_pij?" attribute.
